File: Chapter 04/Python/4.4 Anti-Patterns/Before-FatController_AnemicDomain/my_db_context.py
"""
INFRASTRUCTURE LAYER: DATABASE.
ARCHITECTURE NOTE: This represents a low-level detail 
that the high-level Controller should not be creating.
"""
class MyDbContext:
    def save(self, order):
        print(f"  [DB] Persistence: Saving Order {order.id}")

    @property
    def orders(self):
        # Simulating a collection
        return self

    def add(self, order):
        order.id = 999 # Simulating identity generation
        pass

    def commit(self):
        print("  [DB] Transaction Committed.")

File: Chapter 04/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter04/4.4 Anti-Patterns/Before-FatController_AnemicDomain/MyDbContext.java
package com.grokkingsoftwarearchitecture.chapter04;

/**
 * INFRASTRUCTURE LAYER: DATABASE.
 * ARCHITECTURE NOTE: In the 'Before' state, the Controller 
 * manages the lifecycle of this class (New is Glue).
 */
public class MyDbContext {
    public void save(Order order) {
        System.out.println("  [DB] Persistence logic: Saving Order " 
            + order.id + " to SQL.");
    }
}